src/trio/_core/_run.py

```diff
@@ -734,13 +750,48 @@ def deadline(self) -> float:
         this can be overridden by the ``deadline=`` argument to
         the :class:`~trio.CancelScope` constructor.
         """
+        if self._is_relative is True:
+            raise RuntimeError(
+                "unentered relative cancel scope does not have an absolute deadline",
+            )
```

Comment:
We could make this a bit smoother: emit a deprecation warning, and then return current_time() + relative_deadline